### PR TITLE
Add TargetSource headerVisibility

### PR DESCRIPTION
**Pull request by yonaskolb**
Alternative to #187

This adds an optional `headerVisibility` property to `TargetSource`.

It also changes the default visibility of headers from Public to Project. This matches the default that Xcode has for header files. (Interestingly Xcode also doesn't explicitly write the visibility if it set to  Project) 
This change isn't required for this PR, and is potentially breaking. Does anyone have any thoughts on changing the default?

### CODE REVIEW

File: Docs/ProjectSpec.md [hunk 1]
```diff
@@ -238,6 +238,10 @@ A source can be provided via a string (the path) or an object of the form:
 	- `file`: a file reference with a parent group will be created (Default for files or directories with extensions)
 	- `group`: a group with all it's containing files. (Default for directories without extensions)
 	- `folder`: a folder reference.
+- [ ] **headerVisibility**: **String** The visibility of any headers. This defaults to `project`, but can be either:
+	- `public`
+	- `private`
+	- `project`
```
Comment (toshi0383) [inline Docs/ProjectSpec.md]: Maybe we can allow upper-case at first letter. `Public` `Private` `Project`

Comment (yonaskolb) [reply Docs/ProjectSpec.md]: Yeah I guess the decoding could lowercase it first. I'd want the documented way to be lowercase though to match the rest of the spec

**Comment by toshi0383** [comment]
> it worked perfectly fine for my app

Oh maybe I just had dirty workspace. For embedded frameworks, maybe it should be `public` by default? 🤔

screen shot 2018-02-27 at 17 04 39

**Comment by yonaskolb** [comment]
Yeah keeping public is the easiest thing to do, that won't break existing builds, and makes it so you probably don't have to do anything. You would be leaking some internal headers though

**Comment by toshi0383** [comment]
Xcode makes the umbrella header public when you add "Cocoa Touch Framework" target, so I guess we can go either way.
I would keep current behavior, but I'd like to hear what others think.

**Comment by yonaskolb** [comment]
Changed default back to `public`

**Comment by iamtomcat** [comment]
The default isn't reflected in the docs.

**Comment by yonaskolb** [comment]
Good pickup @iamtomcat. Fixed in master 👍

**Comment by iamtomcat** [comment]
@yonaskolb No problem.

**Pull request merged**